django/urls/conf.py

```diff
@@ -57,6 +57,12 @@ def include(arg, namespace=None):
 def _path(route, view, kwargs=None, name=None, Pattern=None):
     from django.views import View
 
+    if kwargs is not None and not isinstance(kwargs, dict):
+        raise TypeError(
+            f'Keyword arguments must be dict, not {type(kwargs).__name__}. '
+            f'Hint: You are probably trying to pass the URL name as a positional argument.'
```

Comment:
IMO hint is not necessary, we also don't allow for calling functions in f-strings: 
```suggestion
            f'kwargs argument must be a dict, but got {kwargs.__class__.__name__}.'
```

Reply:
One of the main points of the ticket is displaying a friendly message to the developer, I dont think he would get it right away without the hint. I'll add a conditional for only showing the hint when `name` is also None. Then we can be almost 100% sure that he is just forgetting to pass the name as a kwarg.

Reply:
I don't think it needs any more logic.

Reply:
Okay then, I'll follow your suggestions.